sw/device/lib/pinmux.c

```diff
@@ -11,19 +11,32 @@
 
 #define PINMUX0_BASE_ADDR TOP_EARLGREY_PINMUX_AON_BASE_ADDR
 
+// TODO: autogenerated constants for this will be available once
+// the pinmux datastructure has been overhauled.
+#define NUM_GPIO 32
+#define UART_RX_PERIPH_IDX 42
+#define UART_TX_PERIPH_IDX 42
+#define MIO_32_IOR2_IDX 32
+#define MIO_33_IOR3_IDX 33
```

Comment:
Follow-up question: I can see where 32 and 33 come from. But out of curiosity, where does the 42 come from? Is there a big, earth-sized machine somewhere that spit this out? ;-)

Reply:
yes, the earth sized machine is called `topgen` and the index itself is a function of the definition order inside the `mio_modules` list in `top_*.hjson` and the order inside the IP-specific available IO lists.

it looks like this is also the cause for the incorrect pinmux config on the englishbreakfast top.
i.e.,  since the enumeration order of the `gpio` and `uart0` modules in the `mio_modules` list was not the same. 

I fixed the order now, and slightly changed topgen such that it keeps `inouts` and `inputs`/`outputs` in-order instead of stacking all `inouts` first.

Reply:
Ah I see, thanks for clarifying and fixing/stabilizing the issue! One day `topgen` will conquer the world. Could it be the OT is just about coming up with `topgen`?